### PR TITLE
Box workbook and rules

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Box workbook
  - add Box detection rules
  - add Box hunting queries

### CODE REVIEW

**Comment by preetikr** [comment]
@shainw - please review the hunting queries and detections + parser - thanks.
@v-jayakal / @nazang - please review the workbook - thanks.

**Comment by preetikr** [comment]
@vu-socprime  - please fix the validation errors and also handle branch conflicts. thanks.

**Comment by v-admahe** [comment]
@vu-socprime could you please update parser link in workbook to "https://aka.ms/sentinel-BoxDataConnector-parser".

Thanks,
Aditi Maheshwari

**Comment by vu-socprime** [comment]
> https://aka.ms/sentinel-BoxDataConnector-parser

@v-admahe done

**Comment by vu-socprime** [comment]
@preetikr This PR has two errors:
1. _"Template doesn't have valid connectorIds. If a new connector is used and already configured in the Portal, please add it's Id to the list in 'ValidConnectorIds.json' file."_ - This error should be ok after packaging because field requiredDataConnectors has correct data connector ID value, so validation can be considered as successful.
2. _"Can't locate preview image file InfobloxCDCB1TDBlack.png under the Workbooks/Images/Preview directory"_ - Infoblox connecter has this error, so it must be fixed and merged to this branch.

**Comment by preetikr** [comment]
@vu-socprime - For point 1, Yes, Box data connector ID seems fine in the analytic and hunting queries as these are being packaged as Solutions along with the data connector so I'd consider the failed test for this scenario as false positive. 
@shainw, please let us know if you agree and please also review the analytic, hunting queries and parser in this PR. 

@v-jayakal - Can we please fix the validation process so that if the data connector ID is mentioned in analytic, hunting queries, etc. it doesn't flag failures? Also on point 2 the error is flagged on Infoblox workbook missing files. Infoblox is being packaged as a Solution too so can you please remove the metadata reference to Infoblox InfobloxCDCB1TDBlack from the workbook metadata file? 

Thanks,
Preeti

**Review by v-jayakal — APPROVED** [review]

**Comment by v-jayakal** [comment]
Workbook changes are approved.

**Comment by v-rucdu** [comment]
> Infoblox
@preetikr  - 
1. To fix the validation error we have added connectorId in ValidConnectorIds.json file. Have dropped an email to discuss further about fixing the validation process.
2. Removed the metadata reference to Infoblox InfobloxCDCB1TDBlack from the workbook metadata file

File: Detections/Box/BoxInactiveUserLogin.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+query: |
+  let lbperiod_start = 90d;
```
Comment (shainw) [inline Detections/Box/BoxInactiveUserLogin.yaml:17]: 14d max lookback recommended, depending on the size of the data, this query lookback could cause performance issues for the customer who implement it.

File: Detections/Box/BoxItemSharedToExternalUser.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
+query: |
+  BoxEvents
+  | where EventType == 'COLLABORATION_INVITE'
+  | extend corp_domain = extract(@'@(.*)', 1, SrcUserUpn)
+  | extend accessibleby_domain = tolower(extract(@'@(.*)', 1, AccessibleByLogin))
+  | where corp_domain != accessibleby_domain
```
Comment (shainw) [inline Detections/Box/BoxItemSharedToExternalUser.yaml:21]: recommend !~ unless you want exact char match.

File: Detections/Box/BoxMultipleItemsDeletedByUser.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+id: 1b212329-6f2c-46ca-9071-de3464f3d88d
+name: Box - Many items deleted by user
+description: |
+  'Detects when a user deletes many items in short period of time.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 5m
```
Comment (shainw) [inline Detections/Box/BoxMultipleItemsDeletedByUser.yaml:10]: 5m is not needed here.  You are looking at the last 1h every 5m.  We don't recommend 5m, but if it is needed, then queryPeriod should also be 5m if the goal is greater than 100 deleted items in 5m.  If it is, then set to 1h frequency and use bin(TimeGenerated, 5m) which will give you the count every 5m to check the threshold match but only run hourly.

File: Detections/Box/BoxSensitiveFile.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+tactics:
+  - InitialAccess
+  - Persistence
+query: |
+  BoxEvents
+  | where SourceItemName == 'id_rsa' or SourceItemName contains 'password' or SourceItemName contains 'key' or SourceItemName contains '_key' or SourceItemName contains '.ssh' or SourceItemName endswith '.npmrc' or SourceItemName endswith '.muttrc' or SourceItemName contains 'config.json' or SourceItemName contains '.gitconfig' or SourceItemName endswith '.netrc' or SourceItemName endswith 'package.json' or SourceItemName endswith 'Gemfile' or SourceItemName endswith 'bower.json' or SourceItemName endswith 'config.gypi' or SourceItemName endswith 'travis.yml' or SourceFileName == 'id_rsa' or SourceFileName contains 'password' or SourceFileName contains 'key' or SourceFileName contains '_key' or SourceFileName contains '.ssh' or SourceFileName endswith '.npmrc' or SourceFileName endswith '.muttrc' or SourceFileName contains 'config.json' or SourceFileName contains '.gitconfig' or SourceFileName endswith '.netrc' or SourceFileName endswith 'package.json' or SourceFileName endswith 'Gemfile' or SourceFileName contains 'bower.json' or SourceFileName contains 'config.gypi' or SourceFileName contains 'travis.yml'
```
Comment (shainw) [inline Detections/Box/BoxSensitiveFile.yaml:19]: contains is not recommended, try using `has_any` - https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/datatypes-string-operators

File: Detections/Box/BoxUserLoginAsAdmin.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+query: |
+  let lbperiod_start = 60d;
```
Comment (shainw) [inline Detections/Box/BoxUserLoginAsAdmin.yaml:17]: 60d is not recommended, 14d max recommended.

File: Detections/Box/BoxUserRoleChangedToOwner.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+query: |
+  let lbperiod = 60d;
```
Comment (shainw) [inline Detections/Box/BoxUserRoleChangedToOwner.yaml:17]: 60d is not recommended, 14d max recommended.

File: Hunting Queries/Box/BoxAdminIpAddress.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+tactics:
+  - InitialAccess
+  - PrivilegeEscalation
+query: |
+  BoxEvents
+  | where TimeGenerated > ago(60d)
```
Comment (shainw) [inline Hunting Queries/Box/BoxAdminIpAddress.yaml:15]: 60d is not recommended, 30d max recommended for hunting query defaults.

File: Detections/Box/BoxUserRoleChangedToOwner.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  let lbtime = 1h;
+  BoxEvents
+  | where TimeGenerated between (ago(lbperiod) .. ago(lbtime))
+  | where EventType == 'COLLABORATION_INVITE'
```
Comment (shainw) [inline Detections/Box/BoxUserRoleChangedToOwner.yaml:21]: may want to consider =~.

File: Hunting Queries/Box/BoxSuspiciousFiles.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
...Name contains 'config.json' or SourceItemName contains '.gitconfig' or SourceItemName endswith '.netrc' or SourceItemName endswith 'package.json' or SourceItemName endswith 'Gemfile' or SourceItemName endswith 'bower.json' or SourceItemName endswith 'config.gypi' or SourceItemName endswith 'travis.yml' or SourceItemName endswith '.ps1' or SourceItemName endswith '.bat' or SourceItemName endswith '.scr' or SourceItemName endswith '.sh' or SourceItemName endswith '.exe' or SourceFileName == 'id_rsa' or SourceFileName contains 'password' or SourceFileName contains 'key' or SourceFileName contains '_key' or SourceFileName contains '.ssh' or SourceFileName endswith '.npmrc' or SourceFileName endswith '.muttrc' or SourceFileName contains 'config.json' or SourceFileName contains '.gitconfig' or SourceFileName endswith '.netrc' or SourceFileName endswith 'package.json' or SourceFileName endswith 'Gemfile' or SourceFileName contains 'bower.json' or SourceFileName contains 'config.gypi' or SourceFileName contains 'travis.yml' or SourceFileName endswith '.ps1' or SourceFileName endswith '.bat' or SourceFileName endswith '.scr' or SourceFileName endswith '.sh' or SourceFileName endswith '.exe'
```
Comment (shainw) [inline Hunting Queries/Box/BoxSuspiciousFiles.yaml:15]: contains is not recommended, try using `has_any` - https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/datatypes-string-operators

**Review by shainw — CHANGES_REQUESTED** [review]
Several comments within and in general all detection and hunting queries must have relevantTechniques - https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#relevanttechniques.

Also, suggest you make sure you want to use strict matching on strings where you use `==` or `in` instead of `=~` or `in~`.

File: Detections/Box/BoxAbnormalUserActivity.yaml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+query: |
+  let lbperiod_start = 30d;
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Detections/Box/BoxAbnormalUserActivity.yaml]: updated

File: Detections/Box/BoxBinaryFile.yaml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+query: |
+  BoxEvents
+  | where SourceFileName endswith '.exe' or SourceItemName endswith '.exe'
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Detections/Box/BoxBinaryFile.yaml]: updated

File: Detections/Box/BoxAbnormalUserActivity.yaml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Detections/Box/BoxAbnormalUserActivity.yaml]: updated

File: Detections/Box/BoxInactiveUserLogin.yaml (hunk 1 above)
Comment (vu-socprime) [reply Detections/Box/BoxInactiveUserLogin.yaml]: updated

File: Detections/Box/BoxItemSharedToExternalUser.yaml (hunk 2 above)
Comment (vu-socprime) [reply Detections/Box/BoxItemSharedToExternalUser.yaml:23]: updated

File: Detections/Box/BoxMultipleItemsDeletedByUser.yaml (hunk 3 above)
Comment (vu-socprime) [reply Detections/Box/BoxMultipleItemsDeletedByUser.yaml]: updated

File: Detections/Box/BoxSensitiveFile.yaml (hunk 4 above)
Comment (vu-socprime) [reply Detections/Box/BoxSensitiveFile.yaml:20]: updated

File: Detections/Box/BoxUserLoginAsAdmin.yaml (hunk 5 above)
Comment (vu-socprime) [reply Detections/Box/BoxUserLoginAsAdmin.yaml:19]: updated

File: Detections/Box/BoxUserRoleChangedToOwner.yaml (hunk 6 above)
Comment (vu-socprime) [reply Detections/Box/BoxUserRoleChangedToOwner.yaml]: updated

File: Hunting Queries/Box/BoxAdminIpAddress.yaml (hunk 7 above)
Comment (vu-socprime) [reply Hunting Queries/Box/BoxAdminIpAddress.yaml]: updated

File: Detections/Box/BoxUserRoleChangedToOwner.yaml (hunk 8 above)
Comment (vu-socprime) [reply Detections/Box/BoxUserRoleChangedToOwner.yaml]: updated

File: Hunting Queries/Box/BoxSuspiciousFiles.yaml (hunk 9 above)
Comment (vu-socprime) [reply Hunting Queries/Box/BoxSuspiciousFiles.yaml:17]: updated

File: Detections/Box/BoxInactiveUserLogin.yaml [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+name: Box - Inactive user login
+description: |
+  'Detects user login after long inactivity period.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
```
Comment (shainw) [inline Detections/Box/BoxInactiveUserLogin.yaml:11]: change to 14d as this i the lookback period.

File: Detections/Box/BoxItemSharedToExternalUser.yaml (hunk 2 above)
Comment (shainw) [reply Detections/Box/BoxItemSharedToExternalUser.yaml:23]: doesn't look like this was updated.

File: Detections/Box/BoxSensitiveFile.yaml (hunk 4 above)
Comment (shainw) [reply Detections/Box/BoxSensitiveFile.yaml:20]: does not look like this was updated

File: Detections/Box/BoxUserLoginAsAdmin.yaml (hunk 5 above)
Comment (shainw) [reply Detections/Box/BoxUserLoginAsAdmin.yaml:19]: does not look like this was updated

File: Detections/Box/BoxUserRoleChangedToOwner.yaml [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+name: Box - User role changed to owner
+description: |
+  'Detects when user collaboration role is changed to owner.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+queryFrequency: 1h
+queryPeriod: 1h
```
Comment (shainw) [inline Detections/Box/BoxUserRoleChangedToOwner.yaml:11]: change to 14d to match the lookback

File: Hunting Queries/Box/BoxSuspiciousFiles.yaml (hunk 9 above)
Comment (shainw) [reply Hunting Queries/Box/BoxSuspiciousFiles.yaml:17]: does not look like this was updated

**Review by shainw — CHANGES_REQUESTED** [review]
A few things left to fix up.

File: Detections/Box/BoxItemSharedToExternalUser.yaml (hunk 2 above)
Comment (vu-socprime) [reply Detections/Box/BoxItemSharedToExternalUser.yaml:23]: rule fixed - added toLower() function

File: Detections/Box/BoxSensitiveFile.yaml (hunk 4 above)
Comment (vu-socprime) [reply Detections/Box/BoxSensitiveFile.yaml]: has_any is not applicable in this case. If we use has_any the rule will not cover all desired events(parts of file name)

File: Hunting Queries/Box/BoxSuspiciousFiles.yaml (hunk 9 above)
Comment (vu-socprime) [reply Hunting Queries/Box/BoxSuspiciousFiles.yaml]: has_any is not applicable in this case. If we use has_any the rule will not cover all desired events(parts of file name)

File: Detections/Box/BoxUserLoginAsAdmin.yaml (hunk 5 above)
Comment (vu-socprime) [reply Detections/Box/BoxUserLoginAsAdmin.yaml]: fixed

File: Detections/Box/BoxInactiveUserLogin.yaml (hunk 13 above)
Comment (vu-socprime) [reply Detections/Box/BoxInactiveUserLogin.yaml]: fixed

File: Detections/Box/BoxUserRoleChangedToOwner.yaml (hunk 14 above)
Comment (vu-socprime) [reply Detections/Box/BoxUserRoleChangedToOwner.yaml]: fixed

**Comment by vu-socprime** [comment]
> A few things left to fix up. Please be sure to merge master back to your branch, it is now out of sync.

All things are fixed. Branch master merged.

**Review by shainw — APPROVED** [review]

**Comment by shainw** [comment]
@vu-socprime - thanks for all the work here and iterations, great to have these finalized!  Shain

**Pull request merged**